### PR TITLE
fixed clicks for single choice tree

### PR DESCRIPTION
**Pull request by AlbinaMakisheva**

### CODE REVIEW

**Review by mirekys — CHANGES_REQUESTED** [review]
Screenshot 2024-06-07 at 16 11 48

Confirm button has no use here (other than that it just produces an error when clicked), remove it

**Comment by mirekys** [comment]
Screenshot 2024-06-07 at 16 30 36
Padding bettween checkbox & text dissapeared

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx [hunk 1]
```diff
@@ -318,7 +326,11 @@ export const TreeSelectFieldModal = ({
                 <Button
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx:326]: Semantically wrong element, shouldn't be a button, this is a List.Item & only button this list item should contain is the ">" to open children column. Correct DOM should look like (not exact, you can possibly use List.Content, List.Icon or List.Header inside as per SUI docs):

```
<List.Item>
  <Checkbox /> // if multiple
  {option.hierarchy.title[0]}
  <Button> // if parent node
    <Icon name="angle right" color="black" />
  </Button>
</List.Item>
```

Comment (mirekys) [reply oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx:326]: Using the correct elements would be much easier to do, than to convince buttons to look like list items and don't break the other buttons appearance at the same time ;)

**Review by mirekys — CHANGES_REQUESTED** [review]

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx [hunk 2]
```diff
@@ -285,62 +286,72 @@ export const TreeSelectFieldModal = ({
     }
   };
 
+  useEffect(() => {
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx:289]: Can we do it without resorting to useEffect? When in multiselect mode, this effect is completely unnecessarily registered & evaluated everytime someone selects something...

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx [hunk 3]
```diff
@@ -379,15 +389,15 @@ export const TreeSelectFieldModal = ({
             </Grid>
           </div>
         </Grid>
-      </ModalContent>
-      <ModalActions>
-        <Grid.Row className="gapped">
+      </ModalContent>{" "}
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyTreeSelectField/TreeSelectFieldModal.jsx:392]: Why is this space here?

**Review by mirekys — CHANGES_REQUESTED** [review]

**Comment by Ducica** [comment]
@AlbinaMakisheva @mirekys May I please ask for another modification, where the TreeSelect will accept also a prop, that is a function and that will enable to make filtering of all options based on some condition, similar like it is in https://github.com/oarepo/oarepo-vocabularies/blob/bfbd29028d77c623774d06f036897477c2c4a424/oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/LocalVocabularySelectField/LocalVocabularySelectField.jsx#L102 filterFunction prop. The reason is that some fields, for example, resource type, we need to have different options depending if they are located in the main form, or in the relatedItems field. Thanks!

**Comment by mirekys** [comment]
> @AlbinaMakisheva @mirekys May I please ask for another modification, where the TreeSelect will accept also a prop, that is a function and that will enable to make filtering of all options based on some condition, similar like it is in
> 
> https://github.com/oarepo/oarepo-vocabularies/blob/bfbd29028d77c623774d06f036897477c2c4a424/oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/LocalVocabularySelectField/LocalVocabularySelectField.jsx#L102
> 
> filterFunction prop. The reason is that some fields, for example, resource type, we need to have different options depending if they are located in the main form, or in the relatedItems field. Thanks!

Yes this `filterFunction` prop is needed also for restoration repo for pre-filtering of options based on object's primary material. E.g. if object is from "sklo", then all the `restorationMethods` also needs to be prefiltered by that. Isn't the functionality already somewhere there @AlbinaMakisheva ?

**Review by mirekys — CHANGES_REQUESTED** [review]
Clicking on parent selects nothing:
https://github.com/oarepo/oarepo-vocabularies/assets/646743/65e90bde-9e19-4450-b68c-44b20a03f215

Additionally theres no pointer cursor indicating that item is clickable, theres no hover state

**Review by mirekys — COMMENTED** [review]
On mobile it is impossible to see 3rd level of tree (missing horizontal scroll container if modal content overflows screen)
Screenshot 2024-06-19 at 17 30 00

**Review by mirekys — CHANGES_REQUESTED** [review]
Currently selected value not reflected by modal (it should be highlighted as selected)
https://github.com/oarepo/oarepo-vocabularies/assets/646743/85ef41e4-bc97-4f4e-9d7a-c46338915e74

or checkboxes checked in multi-selection mode:

https://github.com/oarepo/oarepo-vocabularies/assets/646743/2c285365-681b-4532-b2a8-5529f25e2704

**Comment by mirekys** [comment]
> Clicking on parent selects nothing: https://github.com/oarepo/oarepo-vocabularies/assets/646743/65e90bde-9e19-4450-b68c-44b20a03f215
> 
> Additionally theres no pointer cursor indicating that item is clickable, theres no hover state

Still no pointer cursor to indicate clickability, i get `I` cursor when hovering over an item

**Comment by mirekys** [comment]
> Currently selected value not reflected by modal (it should be highlighted as selected) https://github.com/oarepo/oarepo-vocabularies/assets/646743/85ef41e4-bc97-4f4e-9d7a-c46338915e74

In single-value mode, currently selected value is still not shown/distinguished from others when modal opens

**Review by mirekys — APPROVED** [review]

**Pull request merged**